### PR TITLE
Analytics event: SELECT_EXTERNAL_SOURCE

### LINKED ISSUE #1076: Analytics event: `SELECT_EXTERNAL_SOURCE`
## Description

The goal is to 

- add the type for event `SELECT_EXTERNAL_SOURCE` to the `Events` type.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/types/analytics.ts#L12

- fire this event from the appropriate location in the codebase using the `useAnalytics` composable.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/composables/use-analytics.ts#L11

## Reference

https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/rfcs/analytics/20221006-implementation_plan_frontend.md?plain=1#L270

## Additional information

- [RFC](https://github.com/WordPress/openverse/blob/main/rfcs/analytics/20221006-implementation_plan_frontend.md#events)
- [Example usage in `VHomeGallery.vue`](https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/components/VHomeGallery/VHomeGallery.vue#L128-L134)

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1076 by @dhruvkb

## Description

This PR adds `SELECT_EXTERNAL_SOURCE` event that is triggered when the user clicks on one of the `external source` buttons inside the External sources popover/modal below the search results or on one of the buttons on the "No results" page.
The buttons are also updated to use the new variants and sizes.

## Testing Instructions

Run analytics and the frontend dev server. Click on one of the external sources below the search results or search for "nonexistentquery" and click on one of the buttons on the "No results" page.
You should see the event in Plausible.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — COMMENTED** [review]
I think we should also include which part of the UI the external source was selected from in something like a `uiComponent` param.

**Comment by zackkrida** [comment]
@obulat @dhruvkb regarding tracking which part of the UI is calling the event, I have a related `useComponentName` composable in one of my event PRs that might be useful:

https://github.com/WordPress/openverse/blob/add/events-change-content-type/frontend/src/composables/use-component-name.ts

File: frontend/src/components/VHeader/VHeaderMobile/VContentSettingsButton.vue [hunk 1]
```diff
@@ -22,7 +22,7 @@
   </VSearchBarButton>
 </template>
 <script lang="ts">
-import VSearchBarButton from "~/components/VHeader/VHeaderMobile/VSearchBarButton.vue"
+import VSearchBarButton from "../../../components/VHeader/VHeaderMobile/VSearchBarButton.vue"
```
Comment (obulat) [inline frontend/src/components/VHeader/VHeaderMobile/VContentSettingsButton.vue:25]:
```suggestion
import VSearchBarButton from "~/components/VHeader/VHeaderMobile/VSearchBarButton.vue"
```

**Review by zackkrida — APPROVED** [review]
Perfect! LGTM.

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
I don't understand the goal behind the `url` property (ambiguous name aside). Here are my two testing examples:

image

Stored in this way, I don't know what we'll be able to use these for. For one, it isn't easy to query any information against this field that isn't _easier_ elsewhere (specifically query and name). On top of that, it effectively duplicates the name and the query.

My question boils down to: what is the goal for the `url` payload data that isn't covered by the other payload items?

**Review by dhruvkb — APPROVED** [review]
Agree with changes suggested by Sara. Everything else LGTM.

File: frontend/src/types/analytics.ts [hunk 2]
```diff
@@ -85,6 +85,24 @@ export type Events = {
...
+   * Description: When the user chooses an external source from the dropdown of external sources
+   * Questions:
+   *   - Which external sources are most popular? This could drive inclusion in Openverse.
+   *   - Are certain media types more popular externally?
+   */
+  SELECT_EXTERNAL_SOURCE: {
+    /** The name of the external source */
+    name: string
+    /** The full URL of the source */
+    url: string
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/src/types/analytics.ts]: I removed the `url` entirely, as it does not provide much useful non-duplicated information.

File: frontend/test/playwright/e2e/external-sources.spec.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    query: string
+  } = { name: "", url: "", mediaType: "", query: "" }
+  page.on("request", (req) => {
+    if (req.method() === "POST") {
+      const requestData = req.postDataJSON()
+      if (requestData?.n == "SELECT_EXTERNAL_SOURCE") {
+        eventData = JSON.parse(requestData?.p)
+      }
+    }
+  })
```
[the remark this answers is not in the document]
Comment (obulat) [reply frontend/test/playwright/e2e/external-sources.spec.ts:18]: Issue created: https://github.com/WordPress/openverse/issues/1992

**Pull request merged**